scripts/externalROMloader.py

```diff
@@ -0,0 +1,188 @@
+  arguments = sys.argv
+  if len(arguments) == 2:
+    # we expect an input file
+    if not os.path.exists(arguments[1]):
+      raise IOError('Not found input file "' + str(os.path.basename(arguments[1]) )+'" in location "' +os.path.dirname(arguments[1])+  '"!')
+    if not arguments[1].endswith("xml"):
+      raise IOError('The input file "' + str(arguments[1])+'" is not an XML!')
+    if os.path.exists(arguments[1]):
+      tree = ET.parse(arguments[1])
+      root = tree.getroot()
+      ravenDir     = root.find(".//RAVENdir")
+      pickledFile  = root.find(".//ROMfile")
+      evaluateData = root.find(".//evaluate")
+      inspectROM   = root.find(".//inspect")
+      outputFilename = root.find(".//outputFile")
+      outputFilename = outputFilename.text if outputFilename is not None else '.'.join(os.path.basename(arguments[1]).split(".")[:-1])+"_out.xml"
+      if ravenDir is None:
+        raise IOError('Not found XML node <RAVENdir>!')
+      ravenDir = ravenDir.text
+      if pickledFile is None:
+        raise IOError('Not found XML node <ROMfile>!')
+      if evaluateData is None:
+        raise IOError('Not found XML node <evaluate>!')
+      pickledFile = pickledFile.text
+
+      if evaluateData is None and inspectROM is None:
+        raise IOError('Not found either XML node <evaluate> nor <inspect>! At least one of them needs to be inputted!')
+      if evaluateData is not None:
+        evalDict = {}
+        for child in evaluateData:
+          evalDict[child.tag] = [float(val) for val in child.text.split()]
+      if inspectROM is not None:
+        inspROM = True if inspectROM.text.lower() in ['true','t','yes'] else False
+      else:
+        inspROM = False
+  else:
+    raise IOError('An input file must be provided!')
+  rom = ravenROMexternal(pickledFile, ravenDir)
+  if evaluateData is not None:
+    output = rom.evaluate(evalDict)
+  if inspROM:
+    settings = rom.getInitParams()
+  # create the output
+  outElement = ET.Element(rom.rom.name)
+  if inspROM:
+    sett = ET.Element("settings")
+    for el,val in settings.items():
+      element      = ET.Element(el)
+      element.text = str(np.asarray(val))
+      sett.append(element)
+    outElement.append(sett)
+  if evaluateData is not None:
+    evalData = ET.Element("evaluations")
+    for cnt,out in enumerate(output):
+      realizationData = ET.Element("evaluation", attrib={'realization':str(cnt+1)})
+      for el,val in evalDict.items():
+        element      = ET.Element(el)
+        element.text = str(val[cnt])
+        realizationData.append(element)
+      evalData.append(realizationData)
+      for el,val in out.items():
+        element      = ET.Element(el)
+        element.text = str(val)
+        realizationData.append(element)
+      evalData.append(realizationData)
+    outElement.append(evalData)
+  with open (outputFilename,"w") as fileObject:
+    fileObject.write(minidom.parseString(ET.tostring(outElement)).toprettyxml(indent="  "))
+
+
```

Comment:
Is there a how-to on using this feature in the guide and/or manual?

Reply:
Yes. the documentation has been added in the IOStep section